dot/network/service.go

```diff
@@ -593,7 +593,7 @@ func (s *Service) SendMessage(to peer.ID, msg NotificationsMessage) error {
 }
 
 func (s *Service) GetRequestResponseProtocol(subprotocol string, requestTimeout time.Duration,
-	maxResponseSize uint64) *RequestResponseProtocol {
+	maxResponseSize uint64) RequestMaker {
```

Comment:
Just a question, if the `*RequestResponseProtocol` implements `RequestMaker` why this change is needed? btw, I think that the approach `return structs, receive interfaces` is better to use given that this function will always only return a single type.

Reply:
I agree that `return structs, receive interfaces` is preferable. I changed the return type of this method in the interface `parachain.Network` in order to mock `RequestResponseProtocol`. But without also changing the return type in `network.Service`, that struct no longer implements the interface.